File: akka-benchmark/src/main/java/io/sarl/akka/hierarchy/AkkaHierarchy.java
package io.sarl.akka.hierarchy;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;

import java.io.IOException;

public class AkkaHierarchy {
    public static void main(String[] args) {
        final ActorSystem system = ActorSystem.create("hierarchy-akka");
        try {
            final ActorRef parent = system.actorOf(Parent.props());

            parent.tell(new Parent.Start(), ActorRef.noSender());

            System.out.println(">>> Press ENTER to exit <<<");
            System.in.read();
        } catch (IOException ignored) {
        } finally {
            system.terminate();
        }
    }
}

File: akka-benchmark/src/main/kotlin/io/sarl/akka/hierarchy/AkkaHierarchy.kt
package io.sarl.akka.hierarchy

import akka.actor.ActorRef
import akka.actor.ActorSystem

import java.io.IOException

object AkkaHierarchy {
    @JvmStatic
    fun main(args: Array<String>) {
        val system = ActorSystem.create("hierarchy-akka")
        try {
            val parent = system.actorOf(Parent.props())

            parent.tell(Parent.Start(), ActorRef.noSender())

            println(">>> Press ENTER to exit <<<")
            System.`in`.read()
        } catch (ignored: IOException) {
        } finally {
            system.terminate()
        }
    }
}
